lib/mac/highsierra.js

```diff
@@ -321,8 +321,13 @@ nobleBindings.on('kCBMsgId72', function(args) {
 
   if (args.kCBMsgArgServices) {
     for(var i = 0; i < args.kCBMsgArgServices.length; i++) {
+      var discoveredServiceUuid = args.kCBMsgArgServices[i].kCBMsgArgUUID.toString('hex');
+      if(discoveredServiceUuid.length > 4) {
+        discoveredServiceUuid = discoveredServiceUuid.slice(4, 8);
+      }
+
```

Comment:
seems like this should be a utility function rather than copy/pasted in all the places you're using it.